pykafka/rdkafka/simple_consumer.py

```diff
@@ -0,0 +1,181 @@
+from contextlib import contextmanager
+import logging
+
+from pykafka.simpleconsumer import (
+    SimpleConsumer, ConsumerStoppedException, OffsetType)
+from . import _rd_kafka
+
+
+log = logging.getLogger(__name__)
+
+
+class RdKafkaSimpleConsumer(SimpleConsumer):
+    """A pykafka.SimpleConsumer with librdkafka-based fetchers
+
+    This aims to conform to the SimpleConsumer interface as closely as
+    possible.  There are some notable differences:
+
+    1. rotating over partitions: while message ordering within partitions is
+       conserved (of course!), the order in which partitions are visited will
+       deviate.  In particular, here we may emit more than one message from
+       the same partition before visiting another
+    2. ignores num_consumer_fetchers: librdkafka will typically spawn at least
+       as many threads as there are kafka cluster nodes
+
+    For an overview of how configuration keys are mapped to librdkafka's, see
+    _mk_rdkafka_config_lists.
+    """
+
+    def _setup_fetch_workers(self):
+        brokers = ','.join(map(lambda b: ':'.join((b.host, str(b.port))),
+                               self._cluster.brokers.values()))
+        partition_ids = list(self._partitions_by_id.keys())
+        start_offsets = [
+            self._partitions_by_id[p].next_offset for p in partition_ids]
+        conf, topic_conf = self._mk_rdkafka_config_lists()
+
+        self._rdk_consumer = _rd_kafka.Consumer()
+        log.debug("Configuring _rdk_consumer...")
+        self._rdk_consumer.configure(conf=conf)
+        self._rdk_consumer.configure(topic_conf=topic_conf)
+
+        start_kwargs = {"brokers": brokers,
+                        "topic_name": self._topic.name,
+                        "partition_ids": partition_ids,
+                        "start_offsets": start_offsets}
+        log.debug("Starting _rdk_consumer with {}".format(start_kwargs))
+        self._rdk_consumer.start(**start_kwargs)
+
+    def consume(self, block=True):
+        timeout_ms = self._consumer_timeout_ms if block else 1
+        try:
+            msg = self._rdk_consumer.consume(timeout_ms)
+        except _rd_kafka.ConsumerStoppedException:
+            raise ConsumerStoppedException
+        if msg is not None:
+            # set offset in OwnedPartition so the autocommit_worker can find it
+            self._partitions_by_id[msg.partition_id].set_offset(msg.offset)
+        return msg
+
+    def stop(self):
+        if hasattr(self, "_rdk_consumer") and self._rdk_consumer is not None:
+            # Call _rd_kafka.Consumer.stop explicitly, so we may catch errors:
+            self._rdk_consumer.stop()
+            log.debug("Issued stop to _rdk_consumer.")
+            self._rdk_consumer = None
+        super(RdKafkaSimpleConsumer, self).stop()
```

Comment:
These `super()` calls should probably all include `return` just for maximum futureproofing.